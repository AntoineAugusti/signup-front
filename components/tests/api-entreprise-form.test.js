import {mount, shallow} from 'enzyme'
import ReactTestRenderer from 'react-test-renderer'
import ApiEntrepriseForm from '../api-entreprise-form'
import Services from '../../lib/services'
import localStorage from '../../lib/tests/local-storage' // eslint-disable-line no-unused-vars
import API_ENTREPRISE_FORM_STATE from '../../mock/enrollment-form/api-entreprise-initial-state-enrollment'

require('../../lib/tests/local-storage') // eslint-disable-line import/no-unassigned-import

describe('components | ApiEntrepriseForm', () => {
  describe('constructor', () => {
    it('should initialize state with an empty json', () => {
      const wrapper = shallow(<ApiEntrepriseForm />)
      expect(wrapper.state()).toEqual(API_ENTREPRISE_FORM_STATE)
    })
  })

  describe('render', () => {
    const props = {
      children: '<p>Some html</p>'
    }
    const renderer = ReactTestRenderer.create(<ApiEntrepriseForm {...props} />)
    it('should be defined', () => {
      expect(renderer).toBeDefined()
    })
    it('should match snapshot', () => {
      expect(renderer).toMatchSnapshot()
    })
    describe('form', () => {
      let props
      let wrapper
      it('should have a `<form>` element', () => {
        props = {
          handleSubmit: () => {
          }
        }
        wrapper = shallow(<ApiEntrepriseForm {...props} />)
        expect(wrapper.find('form')).toHaveLength(1)
      })

      describe('handleChange', () => {
        it('should update the state when a value is input', () => {
          wrapper = shallow(<ApiEntrepriseForm />)
          const name = 'Fake Value'
          const textarea = wrapper.find('textarea').at(0)
          textarea.simulate('change', {
            target: {
              name: 'legal1',
              value: name
            }
          })
          expect(
            wrapper.state().legal1
          ).toBe(name)
        })
      })
      describe('handleSubmit', () => {
        it('should call the services with the good parameters', () => {
          const spy = jest.spyOn(Services, 'createUserEnrollment')
          localStorage.setItem('token', 'fakeToken')
          const wrapper = mount(<ApiEntrepriseForm />)
          const componentInitialState = wrapper.state()
          wrapper.find('form').simulate('submit', {preventDefault: jest.fn()})
          expect(spy).toHaveBeenCalledWith(componentInitialState, 'fakeToken')
        })
      })
    })
  })
})
