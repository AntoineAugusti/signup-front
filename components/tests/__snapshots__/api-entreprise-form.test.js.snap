// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | ApiEntrepriseForm render should match snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <h1>
    Demande d'accès à API Entreprise
  </h1>
  <p>
    Pour avoir accès à l'API Entreprise, diffusant des données personnelles, vous devez obtenir un agrément.  L'accès à cette API n'est pour l'instant disponible que si vous êtes:
  </p>
  <ul>
    <li>
      une administration;
    </li>
    <li>
      une entreprise prestataire d'une administration ou ayant une délégation de service public.
    </li>
  </ul>
  <p>
    Pour utiliser API Entreprise, vous devez vous engager à traiter la bonne donnée par le bon agent de la collectivité et informer correctement l’usager, c'est à dire explicitement demander l'accord de l'usager pour récupérer ses données auprès d'un fournisseur.
  </p>
  <h2
    id="identite"
  >
    Identité
  </h2>
  <div
    className="form__group"
  >
    <label
      htmlFor="search-siren"
    >
      Rechercher votre organisme avec son SIREN
    </label>
    <div
      className="search__group"
    >
      <input
        id="search-siren"
        name="enrollment.siren"
        onChange={[Function]}
        type="text"
        value=""
      />
      <button
        aria-label="Recherche"
        className="overlay-button"
        onClick={[Function]}
        type="button"
      >
        <svg
          className="icon icon-search"
          height="100%"
          id="icon-search"
          width="100%"
        >
          <title>
            Rechercher
          </title>
          <path
            d="M15.504 13.616l-3.79-3.223c-0.392-0.353-0.811-0.514-1.149-0.499 0.895-1.048 1.435-2.407 1.435-3.893 0-3.314-2.686-6-6-6s-6 2.686-6 6 2.686 6 6 6c1.486 0 2.845-0.54 3.893-1.435-0.016 0.338 0.146 0.757 0.499 1.149l3.223 3.79c0.552 0.613 1.453 0.665 2.003 0.115s0.498-1.452-0.115-2.003zM6 10c-2.209 0-4-1.791-4-4s1.791-4 4-4 4 1.791 4 4-1.791 4-4 4z"
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="form__group"
  >
    <label
      htmlFor="raison_sociale"
    >
      Raison sociale
    </label>
    <input
      disabled={true}
      id="raison_sociale"
      name="enrollment.raison_sociale"
      onChange={[Function]}
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="form__group"
  >
    <label
      htmlFor="adresse"
    >
      Adresse
    </label>
    <input
      disabled={true}
      id="adresse"
      name="enrollment.adresse"
      onChange={[Function]}
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="form__group"
  >
    <label
      htmlFor="code_naf"
    >
      Code NAF
    </label>
    <input
      disabled={true}
      id="code_naf"
      name="enrollment.code_naf"
      onChange={[Function]}
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="form__group"
  >
    <label
      htmlFor="responsable"
    >
      Responsable
    </label>
    <input
      disabled={true}
      id="responsable"
      name="enrollment.responsable"
      onChange={[Function]}
      type="text"
      value={undefined}
    />
  </div>
  <h3>
    Contacts
  </h3>
  <div
    className="row card-row"
  >
    <div
      className="card"
    >
      <div
        className="card__content"
      >
        <h3>
          Délégué à la protection des données
        </h3>
        <div
          className="card__meta"
        >
          https://www.cnil.fr/fr/designation-dpo
        </div>
        <div
          className="form__group"
        >
          <label
            htmlFor="person_dpo_nom"
          >
            Nom et Prénom
          </label>
          <input
            disabled={false}
            id="person_dpo_nom"
            name="nom"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </div>
        <div
          className="form__group"
        >
          <label
            htmlFor="person_dpo_email"
          >
            Email
          </label>
          <input
            disabled={false}
            id="person_dpo_email"
            name="email"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card__content"
      >
        <h3>
          Responsable de traitement
        </h3>
        <div
          className="form__group"
        >
          <label
            htmlFor="person_responsable_traitement_nom"
          >
            Nom et Prénom
          </label>
          <input
            disabled={false}
            id="person_responsable_traitement_nom"
            name="nom"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </div>
        <div
          className="form__group"
        >
          <label
            htmlFor="person_responsable_traitement_email"
          >
            Email
          </label>
          <input
            disabled={false}
            id="person_responsable_traitement_email"
            name="email"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card__content"
      >
        <h3>
          Responsable métier
        </h3>
        <div
          className="form__group"
        >
          <label
            htmlFor="person_metier_nom"
          >
            Nom et Prénom
          </label>
          <input
            disabled={false}
            id="person_metier_nom"
            name="nom"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </div>
        <div
          className="form__group"
        >
          <label
            htmlFor="person_metier_email"
          >
            Email
          </label>
          <input
            disabled={false}
            id="person_metier_email"
            name="email"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </div>
      </div>
    </div>
    <div
      className="card"
    >
      <div
        className="card__content"
      >
        <h3>
          Responsable technique
        </h3>
        <div
          className="form__group"
        >
          <label
            htmlFor="person_technique_nom"
          >
            Nom et Prénom
          </label>
          <input
            disabled={false}
            id="person_technique_nom"
            name="nom"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </div>
        <div
          className="form__group"
        >
          <label
            htmlFor="person_technique_email"
          >
            Email
          </label>
          <input
            disabled={false}
            id="person_technique_email"
            name="email"
            onChange={[Function]}
            type="text"
            value={undefined}
          />
        </div>
      </div>
    </div>
  </div>
  <h2
    id="demarche"
  >
    Démarche
  </h2>
  <section
    className="information-text"
  >
    <p>
      C'est la raison pour laquelle vous collectez des données personnelles, l'objectif qui est poursuivi par le fichier que vous mettez en place. Par exemple, « télé-procédure permettant aux usagers de demander une aide au paiement de la cantine des collégiens » ou « télé-procédure de demande de bourses de lycée ».
    </p>
    <p>
      Il vous est également demandé de préciser les références du fondement légal de votre droit à demander ces informations (délibération du conseil municipal, décret …) ainsi que les informations relatives à votre téléservice.
    </p>
  </section>
  <div
    className="form__group"
  >
    <label
      htmlFor="intitule_demarche"
    >
      Intitulé
    </label>
    <input
      disabled={false}
      id="intitule_demarche"
      name="enrollment.demarche.intitule"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <div
    className="form__group"
  >
    <label
      htmlFor="description_service"
    >
      Décrivez brièvement votre service ainsi que l‘utilisation prévue des données transmises
    </label>
    <textarea
      disabled={false}
      id="description_service"
      name="enrollment.demarche.description"
      onChange={[Function]}
      rows="10"
      value=""
    />
  </div>
  <div
    className="form__group"
  >
    <label
      htmlFor="fondement_juridique"
    >
      Cadre juridique 
      <i>
        (Indiquez la référence ou l'URL du texte de loi)
      </i>
    </label>
    <input
      disabled={false}
      id="fondement_juridique_demarche"
      name="enrollment.demarche.fondement_juridique"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <h2
    id="donnees"
  >
    Données
  </h2>
  <div
    className="form__group"
  >
    <fieldset
      className="vertical"
    >
      <label>
        Sélectionnez vos jeux de données souhaités
      </label>
      <div
        className="row"
      >
        <div
          className="column"
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entrepriseattestations_agefiph"
              name="enrollment.scopes.attestations_agefiph"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entrepriseattestations_agefiph"
            >
              Attestation AGEFIPH
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_attestations_agefiph"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_attestations_agefiph"
                  name="enrollment.donnees.destinataires.attestations_agefiph"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entrepriseattestations_fiscales"
              name="enrollment.scopes.attestations_fiscales"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entrepriseattestations_fiscales"
            >
              Attestation Fiscale
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_attestations_fiscales"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_attestations_fiscales"
                  name="enrollment.donnees.destinataires.attestations_fiscales"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entrepriseattestations_sociales"
              name="enrollment.scopes.attestations_sociales"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entrepriseattestations_sociales"
            >
              Attestation Sociale
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_attestations_sociales"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_attestations_sociales"
                  name="enrollment.donnees.destinataires.attestations_sociales"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entreprisecertificat_cnetp"
              name="enrollment.scopes.certificat_cnetp"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entreprisecertificat_cnetp"
            >
              Certificat CNETP
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_certificat_cnetp"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_certificat_cnetp"
                  name="enrollment.donnees.destinataires.certificat_cnetp"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entrepriseassociations"
              name="enrollment.scopes.associations"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entrepriseassociations"
            >
              Association
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_associations"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_associations"
                  name="enrollment.donnees.destinataires.associations"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entreprisecertificat_opqibi"
              name="enrollment.scopes.certificat_opqibi"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entreprisecertificat_opqibi"
            >
              Certificat OPQIBI
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_certificat_opqibi"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_certificat_opqibi"
                  name="enrollment.donnees.destinataires.certificat_opqibi"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entreprisedocuments_association"
              name="enrollment.scopes.documents_association"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entreprisedocuments_association"
            >
              Document association
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_documents_association"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_documents_association"
                  name="enrollment.donnees.destinataires.documents_association"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entrepriseetablissements"
              name="enrollment.scopes.etablissements"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entrepriseetablissements"
            >
              INSEE Etablissement
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_etablissements"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_etablissements"
                  name="enrollment.donnees.destinataires.etablissements"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entrepriseentreprises"
              name="enrollment.scopes.entreprises"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entrepriseentreprises"
            >
              INSEE Entreprise
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_entreprises"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_entreprises"
                  name="enrollment.donnees.destinataires.entreprises"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entrepriseextrait_court_inpi"
              name="enrollment.scopes.extrait_court_inpi"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entrepriseextrait_court_inpi"
            >
              Extrait INPI
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_extrait_court_inpi"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_extrait_court_inpi"
                  name="enrollment.donnees.destinataires.extrait_court_inpi"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entrepriseextraits_rcs"
              name="enrollment.scopes.extraits_rcs"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entrepriseextraits_rcs"
            >
              Extrait RCS
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_extraits_rcs"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_extraits_rcs"
                  name="enrollment.donnees.destinataires.extraits_rcs"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entrepriseexercices"
              name="enrollment.scopes.exercices"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entrepriseexercices"
            >
              Exercice
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_exercices"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_exercices"
                  name="enrollment.donnees.destinataires.exercices"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entrepriseliasse_fiscale"
              name="enrollment.scopes.liasse_fiscale"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entrepriseliasse_fiscale"
            >
              Liasse fiscale
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_liasse_fiscale"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_liasse_fiscale"
                  name="enrollment.donnees.destinataires.liasse_fiscale"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entreprisefntp_carte_pro"
              name="enrollment.scopes.fntp_carte_pro"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entreprisefntp_carte_pro"
            >
              Carte Pro FNTP
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_fntp_carte_pro"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_fntp_carte_pro"
                  name="enrollment.donnees.destinataires.fntp_carte_pro"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entreprisequalibat"
              name="enrollment.scopes.qualibat"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entreprisequalibat"
            >
              Certificat Qualibat
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_qualibat"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_qualibat"
                  name="enrollment.donnees.destinataires.qualibat"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entrepriseprobtp"
              name="enrollment.scopes.probtp"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entrepriseprobtp"
            >
              Certificat PROBTP
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_probtp"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_probtp"
                  name="enrollment.donnees.destinataires.probtp"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
          <div>
            <input
              checked={false}
              className="scope__checkbox"
              disabled={false}
              id="checkbox-scope_api_entreprisemsa_cotisations"
              name="enrollment.scopes.msa_cotisations"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="label-inline"
              htmlFor="checkbox-scope_api_entreprisemsa_cotisations"
            >
              Cotisation MSA
            </label>
            <div
              className="scope__destinataire"
            >
              <div
                className="form__group"
              >
                <label
                  htmlFor="destinataire_msa_cotisations"
                >
                  Destinataires 
                  <a
                    href="https://www.cnil.fr/fr/definition/destinataire"
                    rel="noopener noreferrer"
                    target="_blank"
                  >
                    (plus d´infos)
                  </a>
                </label>
                <input
                  disabled={false}
                  id="desinataire_msa_cotisations"
                  name="enrollment.donnees.destinataires.msa_cotisations"
                  onChange={[Function]}
                  type="text"
                  value={undefined}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
  </div>
  <div
    className="form__group"
  >
    <label
      htmlFor="donnees_conservation"
    >
      Conservation des données 
      <i>
        (en mois)
      </i>
    </label>
    <input
      disabled={false}
      id="donnees_conservation"
      name="enrollment.donnees.conservation"
      onChange={[Function]}
      type="number"
      value=""
    />
  </div>
  <div
    className="form__group"
  >
    <label
      htmlFor="donnees_destinataires"
    >
      Destinataires
    </label>
    <input
      disabled={false}
      id="donnees_destinataires"
      name="enrollment.donnees.destinataires"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <h1
    id="convention"
  >
    Convention
  </h1>
  <section
    className="information-text"
  >
    <p>
      Votre raccordement à l'API « Impôt Particulier » nécessite l'acceptation de la convention d'adhésion fixant vos engagements et ceux de la DGFIP et la DINSIC. 
      <br />
       Les liens ci-dessous vous permettront de visualiser la convention type ainsi que ses annexes. 
      <br />
       La convention générée à l'issue de votre demande de raccordement contiendra l'ensemble des éléments propres à votre situation. 
      <br />
       Cette convention sera publiée sur api.gouv.fr et sera accessible via vos identifiants FranceConnect.
    </p>
  </section>
  <iframe
    height="800px"
    src="static/docs/charte-fc.pdf"
    width="100%"
  />
  <div
    className="form__group"
  >
    <input
      checked={false}
      disabled={false}
      id="validation_de_convention"
      name="enrollment.validation_de_convention"
      onChange={[Function]}
      type="checkbox"
    />
    <label
      className="label-inline"
      htmlFor="validation_de_convention"
    >
      Je valide la présente convention et confirme que le DPO de mon organisme est informé de ma demande
    </label>
  </div>
  <div
    className="button-list"
  >
    <button
      className="button"
      id="submit"
      name="subscribe"
      type="submit"
    >
      Soumettre la demande
    </button>
  </div>
</form>
`;
